locals.tf

```diff
@@ -50,6 +50,22 @@ locals {
   log_analytics_tables = ["AKSAudit", "AKSAuditAdmin", "AKSControlPlane", "ContainerLogV2"]
 }
 
+# Helper locals to make the dynamic block more readable
+# There are three attributes here to cater for resources that
+# support both user and system MIs, only system MIs, and only user MIs
 locals {
-  identity_ids = var.identity_ids != null ? var.identity_ids : azurerm_user_assigned_identity.aks[*].id
+  managed_identities = {
+    user_assigned = length(var.managed_identities.user_assigned_resource_ids) > 0 ? {
```

Comment:
You check `.managed_identities.user_assigned_resource_ids` if it is longer than 0. But you use only the first value. Is this your intention ?

Reply:
I think I am wrong, you are using all of them because user_assigned_resource_ids is a list